src/main/java/application/IncompleteSessionController.java

```diff
@@ -0,0 +1,132 @@
+
+    @Value("${commcarehq.host}")
+    private String host;
+
+    private final Log log = LogFactory.getLog(IncompleteSessionController.class);
+
+    @Autowired
+    @Qualifier(value = "migrated")
+    protected FormSessionRepo migratedFormSessionRepo;
+
+    @ApiOperation(value = "Open an incomplete form session")
+    @RequestMapping(value = Constants.URL_INCOMPLETE_SESSION , method = RequestMethod.POST)
+    public NewFormResponse openIncompleteForm(@RequestBody IncompleteSessionRequestBean incompleteSessionRequestBean,
+                                              @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        log.info("Incomplete session request with bean: " + incompleteSessionRequestBean + " sessionId :" + authToken);
+        SerializableFormSession session;
+        restoreFactory.configure(incompleteSessionRequestBean, new DjangoAuth(authToken));
+        try {
+            session = formSessionRepo.findOneWrapped(incompleteSessionRequestBean.getSessionId());
+        } catch(FormNotFoundException e) {
+            session = migratedFormSessionRepo.findOneWrapped(incompleteSessionRequestBean.getSessionId());
+            // Move over to formplayer db
+            formSessionRepo.save(session);
+        }
+        Lock lock = getLockAndBlock(session.getUsername());
+        try {
+            NewFormResponse response = newFormResponseFactory.getResponse(session);
+            log.info("Return incomplete session response: " + response);
+            return response;
+        } finally {
+            lock.unlock();
+        }
+    }
+    @ApiOperation(value = "Get a list of the current user's sessions")
+    @RequestMapping(value = Constants.URL_GET_SESSIONS, method = RequestMethod.POST)
+    public GetSessionsResponse getSessions(@RequestBody GetSessionsBean getSessionRequest,
+                                           @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        log.info("Get Session Request: " + getSessionRequest);
+
+        String username = TableBuilder.scrubName(getSessionRequest.getUsername());
+
+        restoreFactory.configure(getSessionRequest, new DjangoAuth(authToken));
+
+        List<SerializableFormSession> migratedSessions = migratedFormSessionRepo.findUserSessions(
+                getSessionRequest.getUsername());
+
+        List<SerializableFormSession> formplayerSessions = formSessionRepo.findUserSessions(username);
+
+        ArrayList<FormSession> formSessions = new ArrayList<>();
+
+        for (int i = 0; i < formplayerSessions.size(); i++) {
+            formSessions.add(new FormSession(formplayerSessions.get(i)));
+        }
+
+        if (migratedSessions.size() > 0) {
+
+            // First, get the id of every session we  got from the Formplayer repo so we don't process duplicates
+            ArrayList<String> formplayerSessionIds = new ArrayList<>();
```

Comment:
Interesting, I used an ArrayList because I wanted the contains() function. Apparently using an array and Arrays.asList() would be more performant though: http://stackoverflow.com/questions/1128723/how-can-i-test-if-an-array-contains-a-certain-value